ClosedXML/XLHelper_ASF.cs

```diff
@@ -0,0 +1,142 @@
+﻿// Keep this file CodeMaid organised and cleaned
+
+/* ====================================================================
+   Licensed to the Apache Software Foundation (ASF) under one or more
+   contributor license agreements.  See the NOTICE file distributed with
+   this work for additional information regarding copyright ownership.
+   The ASF licenses this file to You under the Apache License, Version 2.0
+   (the "License"); you may not use this file except in compliance with
+   the License.  You may obtain a copy of the License at
+       http://www.apache.org/licenses/LICENSE-2.0
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+==================================================================== */
+
+using System;
+
+namespace ClosedXML.Excel
+{
+    public static partial class XLHelper
+    {
+        /// <summary>
+        /// Creates a valid sheet name, which is conform to the rules.
+        /// Rules:
+        /// never null
+        /// minimum length is 1
+        /// maximum length is 31
+        /// doesn't contain special chars: : 0x0000, 0x0003, / \ ? * ] [
+        /// Sheet names must not begin or end with ' (apostrophe)
+        /// </summary>
+        /// <param name="nameProposal">can be any string, will be truncated if necessary, allowed to be null</param>
+        /// <param name="replaceChar">the char to replace invalid characters.</param>
+        /// <returns>a valid string, "empty" if too short, "null" if null</returns>
+        // This method was ported from the POI project at https://github.com/apache/poi/blob/trunk/src/java/org/apache/poi/ss/util/WorkbookUtil.java
+        public static string CreateSafeSheetName(string nameProposal, char replaceChar = ' ')
+        {
+            if (nameProposal == null)
+            {
+                return "null";
+            }
+            if (nameProposal.Length < 1)
+            {
+                return "empty";
+            }
+            int MaxWorksheetNameCharsCount = 31;
```

Comment:
`MaxWorksheetNameCharsCount` should either to become a constant (and used in `ValidateSheetName` too) or to be renamed to `maxWorksheetNameCharsCount`.